tests/test_inprocess.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (C) 2016 - 2025 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import os
+import subprocess
+from typing import TYPE_CHECKING
+
+import pytest
+
+if TYPE_CHECKING:
+    from ansys.mapdl.core import Mapdl
+
+from conftest import ON_LOCAL
+
+
+class MapdlInProcessRunner:
+    def __init__(self, wdir: str, exec_path: str | None = None) -> None:
+        self.wdir = str(wdir)
```

Comment:
this runner does not use the current pymapdl library within *PYTHON, which somewhat defeats the purpose of the test. The purpose of test is to answer the following question:

Does the *PYTHON feature work as expected using the current pymapdl package (which is the python environment from which pymapdl is installed and used by pytest). In this context, only python3.10 should be used for this kind of test and the python environment used by the test should be used also by *PYTHON.

Reply:
`*PYTHON` only works with Python3.10?? I dont think it is that restrictive is it? @valallansys ??

Reply:
@germa89 I confirm it only works with python 3.10